### ISSUE TITLE
akka.kafka.internal.KafkaConsumerActor logs credentials on debug level

### ISSUE THREAD

**Issue by AvaPL** [akka/alpakka-kafka#1592]
### Versions used 

Akka version: 2.6.20
Akka Stream Kafka version: 3.0.1

### Expected Behavior

Credentials are not logged in the logs.

### Actual Behavior

Credentials from org.apache.kafka.common.security.plain.PlainLoginModule are logged as plaintext.

### Relevant logs

```
Creating Kafka consumer with akka.kafka.ConsumerSettings(properties=(auto.offset.reset,earliest),(bootstrap.servers,kafka:9092),(client.dns.lookup,use_all_dns_ips),(enable.auto.commit,false),(group.id,my-app),(max.poll.records,250),(sasl.jaas.config,org.apache.kafka.common.security.plain.PlainLoginModule required username='FOOBAR' password='FOOBAR';),(sasl.mechanism,PLAIN),(security.protocol,SASL_SSL),keyDeserializer=Some(org.apache.kafka.common.serialization.StringDeserializer@130dc346),valueDeserializer=Some(io.confluent.kafka.serializers.KafkaAvroDeserializer@1dca3f57),pollInterval=50 milliseconds,pollTimeout=50 milliseconds,stopTimeout=0 days,closeTimeout=20 seconds,commitTimeout=15 seconds,commitRefreshInterval=Duration.Inf,dispatcher=akka.kafka.default-dispatcher,commitTimeWarning=1 second,waitClosePartition=500 milliseconds,metadataRequestTimeout=5 seconds,drainingCheckInterval=30 milliseconds,connectionCheckerSettings=akka.kafka.ConnectionCheckerSettings(enable=false,maxRetries=3,checkInterval=15 seconds,factor=2.0),partitionHandlerWarning=5 secondsresetProtectionSettings=akka.kafka.OffsetResetProtectionSettings(enable=false,offsetThreshold=9223372036854775807,timeThreshold=100000 days)enrichAsync=None)
```

**Comment by ennru** [comment]
This is not very good on Alpakka Kafka, but having `PlainLoginModule` print the password in its `toString` method is rather broken, as well.

**Comment by ennru** [comment]
I've now opened #1614 to make sure only non-security affecting properties are logged.

**Comment by ennru** [comment]
The fix for this issue is released with Alpakka Kafka 4.0.2 (the 4.0.1 version had an error during the release).

**Comment by ennru** [comment]
We've now reported this security problem as [CVE-2023-29471](https://www.cve.org/CVERecord?id=CVE-2023-29471).

**Comment by esamson** [comment]
Will the fix be backported to **3.0.x**?

**Comment by johanandren** [comment]
We do not consider this issue critical and are not planning to backport it.
